dynatrace/dynatrace.go

```diff
@@ -52,37 +52,39 @@ func NewExporter(opts Options) (*Exporter, error) {
 
 	client := &http.Client{}
 
+	defaultDimensions := dimensions.MergeLists(
+		oneagentenrichment.GetOneAgentMetadata(),
+		dimensions.NewNormalizedDimensionList(dimensions.NewDimension("dt.metrics.source", "opentelemetry")),
+		dimensions.NewNormalizedDimensionList(opts.DefaultDimensions...),
+	)
+
 	return &Exporter{
-		client: client,
-		opts:   opts,
-		logger: opts.Logger,
+		client:            client,
+		opts:              opts,
+		defaultDimensions: defaultDimensions,
+		logger:            opts.Logger,
 	}, nil
 }
 
 // Options contains options for configuring the exporter.
 type Options struct {
-	URL      string
-	APIToken string
-	Prefix   string
-	Tags     []string
-	Logger   *zap.Logger
+	URL               string
+	APIToken          string
+	Prefix            string
+	DefaultDimensions []dimensions.Dimension
+	Logger            *zap.Logger
 
 	MetricNameFormatter func(namespace, name string) string
 }
 
 // Exporter forwards metrics to a Dynatrace agent
 type Exporter struct {
-	opts   Options
-	client *http.Client
-	logger *zap.Logger
+	opts              Options
+	defaultDimensions dimensions.NormalizedDimensionList
+	client            *http.Client
+	logger            *zap.Logger
 }
 
-var (
-	reNameAllowedCharList = regexp.MustCompile("[^A-Za-z0-9.-]+")
-	maxDimKeyLen          = 100
-	maxMetricKeyLen       = 250
-)
-
 func defaultFormatter(namespace, name string) string {
```

Comment:
Is this still needed?

Reply:
It's used by the custom formatter option. The user can provide a custom formatter if they want. Removing it would be breaking, but I can't remember where it or the requirement came from in the first place